src/containers/SignUpScreen/index.js

```diff
@@ -2,36 +2,45 @@ import React, { memo, useCallback } from 'react';
 import { Text, View, Button } from 'react-native';
 import { useDispatch } from 'react-redux';
 import { object } from 'prop-types';
+import { isEmpty } from 'lodash';
 
 import SignUpForm from 'components/user/SignUpForm';
 import { signUp } from 'actions/userActions';
+import useSessionChangeEffect from 'hooks/useSessionChangeEffect';
 import translate from 'utils/i18n';
+import { MAIN_SCREEN } from 'constants/screens';
 import styles from './styles';
 
-const SignUpScreen = ({ navigation }) => {
+const SignUpScreen = memo(({ navigation }) => {
   const dispatch = useDispatch();
 
   const signUpRequest = useCallback(user => dispatch(signUp(user)), [dispatch]);
 
+  useSessionChangeEffect(
+    user => {
+      const userLoggedIn = !isEmpty(user);
+      if (userLoggedIn) {
+        navigation.navigate(MAIN_SCREEN);
+      }
+    },
+    [navigation]
+  );
+
   return (
     <View style={styles.container}>
       <Text style={styles.welcome}>{translate('SIGN_UP.title')}</Text>
       <SignUpForm onSubmit={signUpRequest} />
-      <Button title={translate('SIGN_IN.title')} onPress={navigation.pop} />
+      <Button title={translate('SIGN_IN.title')} onPress={() => navigation.goBack()} />
```

Comment:
```suggestion
      <Button title={translate('SIGN_IN.title')} onPress={navigation.goBack} />
```

Reply:
Tried it and it does not work, not sure why though 🤔.

Reply:
Without the arrow function, the function goBack is probably executing in the event handler's context and not in the navigation's context. But who knows 🤔